oauth2/oauth2-core/src/test/java/org/springframework/security/oauth2/core/ClaimAccessorTests.java

```diff
@@ -93,6 +93,31 @@ public void getClaimAsInstantWhenDoubleTypeSecondsThenReturnInstant() {
 				expectedClaimValue.minusSeconds(1), expectedClaimValue.plusSeconds(1));
 	}
 
+	// gh-6187
+	@Test
+	public void getClaimAsInstantWhenISODateStringThenReturnInstant() {
+		Instant expectedClaimValue = Instant.parse("2011-12-03T09:15:30Z");
+		String claimName = "isoDate";
+
+		this.claims.put(claimName, "2011-12-03T09:15:30Z");
+		assertThat(this.claimAccessor.getClaimAsInstant(claimName)).isBetween(
+				expectedClaimValue.minusSeconds(1), expectedClaimValue.plusSeconds(1));
+		this.claims.put(claimName, "2011-12-03T10:15:30+01:00");
+		assertThat(this.claimAccessor.getClaimAsInstant(claimName)).isBetween(
+				expectedClaimValue.minusSeconds(1), expectedClaimValue.plusSeconds(1));
+		this.claims.put(claimName, "2011-12-03T10:15:30+01:00[Europe/Paris]");
+		assertThat(this.claimAccessor.getClaimAsInstant(claimName)).isBetween(
+				expectedClaimValue.minusSeconds(1), expectedClaimValue.plusSeconds(1));
+	}
+
+	// gh-6187
+	@Test(expected = IllegalArgumentException.class)
```

Comment:
Please remove `expected = IllegalArgumentException.class` and use `assertThat(this.claimAccessor...` to be consistent with other tests in this class. Also, `assertThat(..)` is the preferred pattern for testing.